### ISSUE TITLE
UniProtKB uses "UniProt" as assignee

### ISSUE THREAD

**Issue by ValWood** [geneontology/go-annotation#1772]
Some MODs use UniProtKB (TAIR, MGI, E.coli and Zfin)
which I believe is the correct database prefix to use?

http://amigo.geneontology.org/amigo/base_statistics

Who to assign to?

**Comment by ValWood** [comment]
I'm going to pick..... @tonysawfordebi

**Comment by ValWood** [comment]
From
https://github.com/geneontology/go-site/issues/504#issuecomment-358104961

**Comment by tonysawfordebi** [comment]
What's the issue here?

When you refer to an entry in the UniProt database, then the correct form of the CURIE to use is UniProtKB:<accession>, e.g., UniProtKB:Q4VCS5

Annotations that are made by UniProt curators - or are based on predictions from UniProt pipelines (e.g., UniRule) - have UniProt in their assigned_by column.

**Comment by ValWood** [comment]
Hi Tony,

So there are 2 valid prefixes for UniprotKB: UniProtKB and UniProt?

Oddly this graph (which I saw yesterday on the stats page, but can no longer see...)
https://github.com/geneontology/go-site/issues/504
has  "UniProt" as well as "UniProtKB" for *experimental* annotations.

We use this  file 
http://www.geneontology.org/doc/GO.xrf_abbs
as the "definitive" source for xrefs

and it only includes "UniProtKB". 

I may have imagined this but I'm sure that we were told at the time of the change that "UniProt:" was deprecated? We changed everything internally at the time, and it isn't in this file any longer?

**Comment by tonysawfordebi** [comment]
No.

As I said, if you want to refer to an entry in the UniProtKB database, then the correct form is UniProtKB:accession

Some groups do use UniProt:accession, in their annotations but that is wrong.

Looking at the graph, it appears to show the number of annotations coming from each group, and in this case "UniProt" is correct; I've no idea which annotation sets have "UniProtKB" in their assigned_by column, but it's definitely not ours. Whichever group is (mis-)attributing annotations to UniProtKB rather than UniProt needs to update their pipelines.

Incidentally, the definitive xrefs source is now https://github.com/geneontology/go-site/blob/master/metadata/db-xrefs.yaml (from which GO.xrf_abbs is ultimately derived), and if you look in there, you'll see that "UniProt" is defined as a synonym for "UniProtKB".

**Comment by ValWood** [comment]
I think we are all just following what we think we ought to do ;)

https://mailman.stanford.edu/pipermail/go-consortium/2005-July/012420.html

So, its UniProtKB for database prefix and UniProt for assigning group. 

I thought "UniProt" as a synonym was temporary  to enable the switch over....

**Comment by ValWood** [comment]
https://mailman.stanford.edu/pipermail/go-consortium/2005-July/012423.html

It's ver confusing! I really thought we weren't supposed to use "UniProt"  at all.

**Comment by tonysawfordebi** [comment]
Blimey - that's dredging up ancient history... That pre-dates my time at the EBI by a few years, so if it is the case that the synonym was introduced as a temporary measure to enable a switch-over, I'd say that more than enough time has elapsed, and the synonym should be removed.

So, just to be absolutely clear, if you want to refer to an entry in the UniProtKB database, the prefix to use is "UniProtKB".

An aside: there was a brief period just after I joined the EBI when the prefix was either "UniProtKB/Swiss-Prot" or "UniProtKB/TrEMBL" to indicate whether we were referring to a reviewed or an unreviewed entry, but that was dropped pretty swiftly, and we reverted to using UniProtKB for everything, and it has been that way for at least the last eight years.

**Comment by ValWood** [comment]
OK so UniprotKB in "assigned by" are from
MGD @ukemi 
zfin @doughowe  
ecoli @
 tair @tberardini

**Comment by tonysawfordebi** [comment]
Thanks for checking - I was just about to hoover up all of the GAFs from the GOC site to see who the culprits were, so you've saved me a job...

**Comment by ValWood** [comment]
- [ ] action  remove UniProt synonym form UniProtKB entry in http://www.geneontology.org/doc/GO.xrf_abbs

**Comment by ValWood** [comment]
;) you owe me one

But you can see who is using UniProt only in  the xrefs column ;)

**Comment by tonysawfordebi** [comment]
I'll remove the synonym from https://github.com/geneontology/go-site/blob/master/metadata/db-xrefs.yaml and that will eventually make its way into GO.xrf_abbs

**Comment by tonysawfordebi** [comment]
Synonym now removed.

**Comment by hdrabkin** [comment]
The "UniProtKB" is erroneously being used for our GOA_human/ISO load, instead of UniProt; I can put a tr in to get that fixed to UniProt easily enough, but I can't say exactly how fast it will get seen to.
@tonysawfordebi , to be clear, UniProt wants column 15 = UniProt

**Comment by tberardini** [comment]
> UniProt is the group; UniProtKB is the database.

Alrighty then, I'll put in the request to change UniProtKB to UniProt for the assigned by.  Will be visible in the next quarterly release.

**Comment by tonysawfordebi** [comment]
@hdrabkin Yes, exactly.

**Comment by mah11** [comment]
Quoth Tony:

> I'll remove the synonym from https://github.com/geneontology/go-site/blob/master/metadata/db-xrefs.yaml ..

It looks like this change has had the opposite of the desired effect: Mike's GAF-checking script now rejects lines with "UniProt" in the assigned_by column, because UniProt is no longer an allowed abbreviation or synonym.

I'll change the PomBase GAF to use UniProtKB in col. 15 until the checking script is updated, and then change back.

**Comment by tonysawfordebi** [comment]
Bugger.

It sounds like what is actually needed is a new entry in db-xrefs for UniProt - completely separate from the existing UniProtKB one - to indicate that it is a legal group, but does not mint any identifiers.

**Comment by tonysawfordebi** [comment]
I've added a separate entry for UniProt, so when that change has percolated through the dense undergrowth of the various release pipelines, Mike's script should once again recognise "UniProt" as a valid Assigned_by namespace.

**Comment by hdrabkin** [comment]
Yep; just checked today's gaf check; all 23000; all of the GOA mouse load that came from UniProt have been stripped!
sigh

**Comment by ValWood** [comment]
He is a radical suggestion...... UniProt could use UniProtKB for everything since they insisted everyone change to this. 

It would  have been much less confusing to have one name, which did not change.  However, since the name *was*  changed,  it would be better to keep the new name for everything rather than deciding to go back to the old name which they told everyone NOT to use, for  a single use case (assignee).

This way only UniProt would need to fix anything?

**Comment by hdrabkin** [comment]
Want to be careful; 
Are any other dbs using a different assigned by (15) vs what they use as an identifier prefix (ie, RGD, vs RGD:, etc.)??

**Comment by hdrabkin** [comment]
Or maybe just leave it the way it was?

**Comment by tonysawfordebi** [comment]
That's effectively what I've done.

UniProt is no longer a synonym for UniProtKB; it now exists in db-xrefs.yaml as a namespace/database in its own right, but, crucially one that doesn't mint identifiers, which means that UniProt:P12345 will not be recognised as a valid identifier, but UniProtKB:P12345 will.

**Comment by mah11** [comment]
No, what Val suggested is _not to use UniProt for anything at all_, and to use UniProtKB in the assigned_by field as well as in identifier prefixes.

**Comment by tonysawfordebi** [comment]
@mah11 I know; I was replying to Harold, but neglected to tag him.

**Comment by mah11** [comment]
I also see that he added quite a bit to his comment in an edit; most of it didn't come through in the email notification. That said, it's not clear what's gained by using two different strings in the different fields, and therefore not clear whether it's actually worth this hassle to accommodate it. In the past we were implicitly told to use "UniProtKB" for all GO annotation purposes, and this ticket is the first explicit announcement to the contrary that I have seen.

**Comment by tonysawfordebi** [comment]
I don't know when that was done - it pre-dates my time on the project - but one thing that I can say before I have to shoot off is that UniProt and UniProtKB are not synonymous; UniProtKB is just one part (albeit a mighty big part) of UniProt.

**Comment by mah11** [comment]
and my only point is that that fact has not been communicated effectively to all corners of the GOC before now (and arguably still not, since many may not see this ticket)

**Comment by ValWood** [comment]
> BTW, ant to be careful;
Are any other dbs using a different assigned by (15) vs what they use as an identifier prefix (ie, RGD, vs RGD:, etc.)??

I think most DBs don't change the assigned by column, we just submit PomBase ones as they come from GOA.
The ones when were UniProtKB were just you guys  who were  using "UniProtKB" everywhere, as you had been told to.

If this is the case, and UniProtKB/UniProt, whatever we should call them used "UniProtKB" everywhere as  UniProtKB/UniProt requested), and you guys did nothing, I think everything would be OK, and a lot easier for everyone else  to remember what  UniProtKB/UniProt is called.

**Comment by hdrabkin** [comment]
We did/do not use "UniProtKB" everywhere. In column 15, we do use UniProt when loading the annotations we get from our GOA load, since UniProt is what is in the column 15 from that load. The error was that UniProtKB was being assigned (incorrectly) to our ISO loads from UniProt (looked into the original docs and have no idea why that was done). I think what @tonysawfordebi  just did to fix will be ok, but we will still have thousands striped until my request makes it through our triage system, which will take time.

**Comment by cmungall** [comment]
Things definitely would be easier if the fated change from UniProt to UniProtKB was never made.

We have the option of a one-time costly change to restore harmony, by going back to UniProt as the identifier prefix, along with the rest of the universe, I made a separate ticket for this. Please use this linked ticket for discussions of a reversion in prefix:
https://github.com/geneontology/go-site/issues/515

Let's use this ticket to talk about a path forward while doing the conservative thing and retaining UniProtKB as identifier prefix.

First, it is true we have no rule that the assigned_by field should match the id prefix. However, we have a _convention_ that these match when the annotation provider is essentially the same as an annotation provider. The problem with conventions is that everyone assumes they are the norm and then confusion arises when they are violated...

Second, we have a distinct way of representing groups that can go in the assigned-by field, which is the groups.yaml file: https://github.com/geneontology/go-site/tree/master/metadata/groups.yaml

We've been overloading the db-xrefs metadata with this information for a while. Now we have the unfortunately confusing situation where we have distinct entries for UniProtKB and UniProt, with the former being for IDs and the latter for assigned_by. This is confusing as the rest of the world uses UniProt as the ID prefix, and sometimes has UniProtKB as a synonym.

I think the immediate confusion is fixed if

 1. we remove UniProt as a standalone entity from db-xrefs.yaml
 2. The protein2go infrastructure uses groups.yaml for metadata about assigned_by rather than db-xrefs.yaml

I realize that option 2 would require some time investment from @tonysawfordebi  - My late apologies for not coordinating this sooner

**Comment by cmungall** [comment]
> I don't know when that was done - it pre-dates my time on the project 
> - but one thing that I can say before I have to shoot off is that 
> UniProt and UniProtKB are not synonymous; UniProtKB is just one part 
> (albeit a mighty big part) of UniProt.

Where does this distinction matter? So long as the local fragments are 
unique within the larger entity and are resolvable by the larger entity, 
then it's safe to use.

I think increased granularity is very important in the context of 
attribution - but here we are already using the larger entity

**Comment by ValWood** [comment]
and  the other "parts" UniParc and UniRef don't make GO annotations

**Comment by ValWood** [comment]
Oh,  @cmungall that's what you are saying, but in a clever way!

**Comment by jmcherry** [comment]
I suggest that the GO.xrf_abbs file be updated, actually the upstream source file, immediately.  I don't want to edit the software as there could be unknown other impacts without testing.

When was UniProt removed from GO.xrf_abbs.  On Friday evenings all GAF files are processed.  If UniProt was removed before that then lines with UniProt: in col8 and UniProt in col15 were removed.  If the change was after Friday then only newly submitted files have been effected.

Here is my analysis of who is using what ID in which column:

[UniProt-gaf-columns.txt](https://github.com/geneontology/go-annotation/files/1657084/UniProt-gaf-columns.txt)

Six groups use UniProt in col8 (GeneDB_Lmajor, GeneDB_Pfalciparum, dictyBase, fb, gramene_oryza, wb)

Five groups use UniProtKB in col15 (ecocyc, mgi, pombase, tair, zfin, goa_pdb)

**Comment by hdrabkin** [comment]
Current mouse  gaf shows we have both in column 15
UniProt, coming from our GOA mouse load 23272  <<<< this is actually what the GOA_mouse file has
UniProtKB: human_to_mouse ISO 23272
For anything that uses an identifier, it's always UniProtKB:12345 (column 8; we can't input anything  different in column 8 for a UniProtKB (killed on tab out)

**Comment by jmcherry** [comment]
I added UniProt back to the GO.xrf_abbs file.  I re-ran filtering script on all GAFs.  This added back 100s of thousands of rows that were deleted last Friday.  

I have turned off filtering for now.  That way in case UniProt is removed again there will be no impact.  I judge that its  better to be a few days out of date that to lose thousands of annotations.  All of the GOA files had all annotations with UniProt in col 15 removed too.  In goa_human alone that was >132K annotations.

Someone please let us all, The Consortium, know what we should do.  Once the plan is in place I'll turn filtering back on.

**Comment by tonysawfordebi** [comment]
@cmungall I agree with your proposal to remove the UniProt entry from db-xrefs.yaml and UniProt into groups.yaml.

If nobody has any objections to this - or can see any downside - that's what I'll do.

**Comment by hdrabkin** [comment]
want to reiterate here.
1. make a decision as to what is in column 15 and make it really clear.
2. allow sufficient time for all dbs to comply before the filtering script is turned back on (we generate the gaf and gpad from the db directly and these scripts will need to be altered and that request has to go through our software triage system. I know that since we get the "UniProt" from the GOA mouse load, time would be needed to have them convert UniProt to UniProtKB also

**Comment by tonysawfordebi** [comment]
I've asked people from UniProt who can actually make decisions to look at & comment on this ticket, but - for now at least - I stand by what I said way back up there ^^^, namely that UniProt is the group, and that's what you will find in the assigned_by column of our annotations, and UniProtKB is the namespace prefix for UniProt protein identifiers. The changes that I was talking about a couple of comments ago (the ones that Chris proposed) merely formalise this and (with any luck) restore the status quo with regard to the results of the filtering that is applied on the GOC site.

**Comment by tberardini** [comment]
I'll put my internal (to TAIR) request for UniProtKB->UniProt changes on hold until a final decision has been reached.

**Comment by mjmartinf** [comment]
Our position is what we currently use in the GOA files and Tony Sawford has explained above. Just to recap:

- The prefix to use when referring to the UniProtKB database is UniProtKB. This is what we use in publications and what we recommend to authors ( https://academic.oup.com/nar/article/45/D1/D158/2605721)
Therefore, UniProtKB should be in the db-xrefs.yaml

- The assign_by group is 'UniProt'. This group provides annotations made by UniProt curators and/or UniProt pipelines (e.g. UniRule) which can be extended.
Therefore, UniProt should be in the groups.yaml

The GOA files already provide this approach so it would be advisable for groups not to change this please.

**Comment by cmungall** [comment]
And just a reminder these are the two separate metadata files we use for identifiers and assigned-by respectively:

 - https://github.com/geneontology/go-site/blob/master/metadata/db-xrefs.yaml
 - https://github.com/geneontology/go-site/blob/master/metadata/groups.yaml

I think we can close this now?

**Comment by tberardini** [comment]
Action:

UniprotKB in "assigned by" are from

- [ ] MGD @ukemi
- [ ] zfin @doughowe
- [ ] ecoli @jimhu-tamu
- [ ] tair @tberardini

These are the groups that need to change the assigned_by from UniProtKB to UniProt

**Comment by hdrabkin** [comment]
This is in our tracking system but has not been prioritized by our software engineers

**Comment by sabrinatoro** [comment]
There is a case in our tracker to deal with this issue.

**Comment by hdrabkin** [comment]
MGI has fixed them all.

**Comment by tberardini** [comment]
TAIR's are fixed.  Should be visible with file to be released on April 1, 2018.

**Comment by sabrinatoro** [comment]
ZFIN has fixed them all, as shown our latest GAF.

**Comment by tberardini** [comment]
TAIR has fixed them all, as shown in our latest GAF.

**Comment by pgaudet** [comment]
According to stats here: https://github.com/geneontology/go-annotation/issues/2072#issuecomment-421101111

EcoCyc, RDG and ZFIN still have annotations with 'UniProtKB' as source, rather than **UniProt**. 
 936 ecocyc.gaf.gz
  53949 rgd.gaf.gz
  3 zfin.gaf.gz

@keseler 
@slaulederkind 
@sabrinatoro 

Can you please check with your software team to correct that ? 

Thanks, Pascale

**Comment by sabrinatoro** [comment]
We have updated our GAF to use UniProt as assigned_by in April 2018.
I checked our latest GAF (date 2018/09/13), and I cannot find any annotations with "UniProtKB" in col15. Could these annotations be coming from somewhere else but the ZFIN GAF (?? how would that be??)
Could you tell me more about these particular annotations, please? (there are only 3) I'll see if I can find them at all.
(@pgaudet @ValWood )

**Comment by ValWood** [comment]
These came from a query Seth did yesterday?
https://github.com/geneontology/go-annotation/issues/2072
weirdly I don't see these in AmiGO, so @kltm  will need to answer this one!

**Comment by kltm** [comment]
@ValWood Okay, to be more specific, my search was for UniProtKB in GAF col 15. Looking at the files that get loaded into AmiGO, the numbers are as reported:

```
zgrep -P '\t\d{8}\tUniProtKB\t' rgd.gaf.gz | wc -l
53937
zgrep -v '^!' rgd.gaf.gz | cut -f 15 | grep UniProtKB | wc -l
53937
```

To see this in AmiGO, is a little harder to see, as only the 'assigned_by' is made available, regards of what file it came from. But if you use the species filter for rat:

http://amigo.geneontology.org/amigo/search/annotation?q=*:*&fq=taxon_subset_closure_label:%22Rattus%20norvegicus%22&sfq=document_category:%22annotation%22

you can then see the UniProtKB count under the "Contributors" filter: 53937

**Comment by ValWood** [comment]
Yes it is the "contributors"  column we are trying to standardize here.
I think the problem (might) be that some groups pull in the UniProt data as the database authority they then become the '3rd party submitter' of the UniProt data, but the filed gets munged....

**Comment by pgaudet** [comment]
Is it the contributors or the database column we are trying to standardize
? The problem I notice is people using UniProtKB:P12345; does the prefix
come from Col 1 or Col 15 ?

Pascale

**Comment by ValWood** [comment]
I think that the prefix should come from column1 (DB) 15. At least that is how I have always understood it.
http://geneontology.org/page/go-annotation-file-gaf-format-21

**Comment by tonysawfordebi** [comment]
@ValWood Spot on. The combination of cols 1 & 2 identify a specific entry in a specific database.

Actually it would make a lot of sense to merge those two columns into a single one, so that things are referred to by simple CURIEs everywhere, but I don't have the energy to open that particular can of worms...

**Comment by ValWood** [comment]
But you could open a ticket in 78? days... :)

**Comment by tonysawfordebi** [comment]
Now there's an idea...

**Comment by ValWood** [comment]
All done, sadly closing
